### PR TITLE
tests: add a load test

### PR DESCRIPTION
**Pull request by tam7t**
Addresses https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/issues/32

### CODE REVIEW

**Review by sbadame — APPROVED** [review]

File: docs/testing-notes.md [hunk 1]
```diff
@@ -1,12 +1,66 @@
...
+## Build and deploy notes
+
+Use [Google Cloud Build](https://cloud.google.com/run/docs/building/containers#building_using) and [Container Registry](https://cloud.google.com/container-registry/docs/quickstart) to build and host the plugin docker image.
 
-* Use [Google Cloud Build](https://cloud.google.com/run/docs/building/containers#building_using) and [Container Registry](https://cloud.google.com/container-registry/docs/quickstart) to build and host the plugin docker image.
 ```shell
 $ export PROJECT_ID=<your gcp project>
 $ gcloud config set project $PROJECT_ID
 $ ./scripts/build.sh
+...
```
Comment (colinman) [inline docs/testing-notes.md:11]: what are these ellipses for?

File: docs/testing-notes.md [hunk 2]
```diff
@@ -1,12 +1,66 @@
...
 ```
-* Deploy the plugin as a DaemonSet to your cluster.
+
+Deploy the plugin as a DaemonSet to your cluster.
+
 ```shell
 $ ./scripts/deploy.sh
+...
```
Comment (colinman) [inline docs/testing-notes.md:18]: and this one

File: docs/testing-notes.md [hunk 3]
```diff
@@ -1,12 +1,66 @@
...
+```shell
+$ CASE=single ./script/load-test.sh
+...
+$ kubectl scale --replicas=100 deployment/test-load-one-secret
+...
+$ kubectl scale --replicas=0 deployment/test-load-one-secret
+...
+```
+
+There are differe options for `CASE`:
```
Comment (colinman) [inline docs/testing-notes.md:44]: s/differe/different

File: docs/testing-notes.md [hunk 4]
```diff
@@ -1,12 +1,66 @@
...
+
+* `CASE=single` - a deployment where the pod references a `SecretProviderClass` with only 1 secret.
+* `CASE=many` - a deployment where the pod references a `SecretProviderClass` with 50 secrets.
+* `CASE=seed` - creates 50 secrets for use with `CASE=many`
```
Comment (colinman) [inline docs/testing-notes.md:48]: Should these just be separate scripts?

```
./seed.sh
./many.sh
```

**Review by colinman — APPROVED** [review]

File: docs/testing-notes.md (hunk 1 above)
Comment (tam7t) [reply docs/testing-notes.md:11]: those represent the 'output' of the command, my markdown linter doesnt like `$` if you dont also put 'output'. I'll just remove both

File: docs/testing-notes.md (hunk 4 above)
Comment (tam7t) [reply docs/testing-notes.md]: Updated to positional arguments instead

**Review by colinman — APPROVED** [review]

**Pull request merged**